vm/src/stdlib/math.rs

```diff
@@ -27,20 +27,35 @@ macro_rules! make_math_func {
 make_math_func!(math_fabs, abs);
 
 fn math_isfinite(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
-    arg_check!(vm, args, required = [(value, Some(vm.ctx.float_type()))]);
-    let value = objfloat::get_value(value).is_finite();
+    if args.args.len() != 1 {
+        return Err(vm.new_type_error(format!(
+            "erf() takes exactly one argument ({} given)",
+            args.args.len()
+        )));
+    }
+    let value = objfloat::make_float(vm, &args.args[0])?.is_finite();
     Ok(vm.ctx.new_bool(value))
 }
 
 fn math_isinf(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
-    arg_check!(vm, args, required = [(value, Some(vm.ctx.float_type()))]);
-    let value = objfloat::get_value(value).is_infinite();
+    if args.args.len() != 1 {
+        return Err(vm.new_type_error(format!(
+            "erf() takes exactly one argument ({} given)",
```

Comment:
The error string contains the wrong function name `erf`. Maybe it makes sense to create a seperate function for this double extraction. The code is now copy pasted about 3 times. Please create a helper function which checks the amount of arguments and extracts the floating point number.

Reply:
I completely agree with you that this copy-paste is horrible solution. I hoped to start a discussion from it. 

I think this helper should be integrated into `arg_check` macro.

Whereve arguments are parsed and one of them is expected to be float it can actually be any other real number type macro should really check `make_float`, otherwise it'll be cpython incompatible

```
In [1]: class A:
   ...:     def __float__(self):
   ...:         return 1.0
   ...:     

In [2]: float(A())
Out[2]: 1.0

In [3]: import math

In [4]: math.log(A())
Out[4]: 0.0

```

So I hope you can give an advice on this one. what'd be the best way to aproach it? 
Should it be just hardcoded into a macro as a special case?


I'm also pretty sure that the same implicit conversion should be done for for `int`


For example in this case I'm passing `True` as a port number for socket and it only fails, because of premissions, (it works from root!).
```
In [1]: import socket
In [3]: s = socket.socket()
In [4]: s.bind(('localhost', True))
---------------------------------------------------------------------------
PermissionError                           Traceback (most recent call last)
<ipython-input-4-ced4611914b1> in <module>()
----> 1 s.bind(('localhost', True))

PermissionError: [Errno 13] Permission denied
```

Reply:
I would not yet place this automatic coercion in the arg_check macro yet. This macro is used really a lot. I suggest to create a helper function like `coerce_into_float` or something like that and place that in `objfloat.rs` for now. We can always extend the `arg_check` macro later when we require this logic in all places.

Basically my doubt is: Are you sure we always need to convert int to float at all function calls?

Reply:
Yes, this is possible. I believe you can set the type to None, and then it will extract the correct amount of arguments into the given rust names.

Reply:
That's great. I'll try to come back with a fix as soon as possible